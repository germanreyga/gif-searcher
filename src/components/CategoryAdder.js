import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdder = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((previousCategories) => [
        inputValue.trim(),
        ...previousCategories,
      ]);

      setInputValue("");
    } else {
      window.alert("You must add a valid name");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="e.j. Hunter X Hunter"
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
      />

      <button type="submit">Add new category</button>
    </form>
  );
};

CategoryAdder.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
