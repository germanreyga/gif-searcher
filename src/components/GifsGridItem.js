import React from "react";

export const GifsGridItem = ({ id, title, url }) => {
  return (
    <div className="gif-card animate__animated animate__fadeInLeft">
      <img src={url} alt={title} />
      <p>{title ? title : `No title for this GIF`}</p>
    </div>
  );
};
