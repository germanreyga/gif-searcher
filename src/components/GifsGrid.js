import React from "react";
import { GifsGridItem } from "./GifsGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifsGrid = ({ category }) => {
  const { data: imgs, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="category-name animate__animated animate__fadeIn">
        {category}
      </h3>
      {loading && (
        <p className="animate__animated animate__flash">"Loading images..."</p>
      )}
      {
        <div className="card-grid">
          {imgs.map((img) => {
            return <GifsGridItem key={img.id} {...img} />;
          })}
        </div>
      }
    </>
  );
};

GifsGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
