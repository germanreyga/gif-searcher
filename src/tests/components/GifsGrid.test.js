import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import { GifsGrid } from "../../components/GifsGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Tesing the <GifsGrid /> component", () => {
  const category = "random";

  test("should show the component correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifsGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show the gif items when the images are loaded with useFetchGifs", () => {
    const gifs = [
      { id: "ABC", url: "https://google.com/i/ds.jpg", title: "any" },
      { id: "ABC2", url: "https://google.com/i/ds.jpg", title: "any" },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifsGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifsGridItem").length).toBe(gifs.length);
  });
});
