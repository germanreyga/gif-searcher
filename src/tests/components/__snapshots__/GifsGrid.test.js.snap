// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tesing the <GifsGrid /> component should show the component correctly 1`] = `
<Fragment>
  <h3
    className="category-name animate__animated animate__fadeIn"
  >
    random
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    "Loading images..."
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tesing the <GifsGrid /> component should show the gif items when the images are loaded with useFetchGifs 1`] = `
<Fragment>
  <h3
    className="category-name animate__animated animate__fadeIn"
  >
    random
  </h3>
  <div
    className="card-grid"
  >
    <GifsGridItem
      id="ABC"
      key="ABC"
      title="any"
      url="https://google.com/i/ds.jpg"
    />
    <GifsGridItem
      id="ABC2"
      key="ABC2"
      title="any"
      url="https://google.com/i/ds.jpg"
    />
  </div>
</Fragment>
`;
